p2p/server.py

```diff
@@ -80,24 +102,41 @@ def __init__(self,
 
     async def receive_handshake(
             self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        await wait_with_token(
+            self._receive_handshake(reader, writer),
+            token=self.cancel_token,
+            timeout=HANDSHAKE_TIMEOUT,
+        )
+
+    async def _receive_handshake(
+            self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter) -> None:
+        self.logger.debug("Receiving handshake...")
         # Use reader to read the auth_init msg until EOF
-        msg = await reader.read(ENCRYPTED_AUTH_MSG_LEN)
+        msg = await wait_with_token(
+            reader.read(ENCRYPTED_AUTH_MSG_LEN),
+            token=self.cancel_token,
+            timeout=REPLY_TIMEOUT,
```

Comment:
Maybe it doesn't make sense to enforce a timeout on every reply here. It should be fine if a peer takes 90% of the overall handshake timeout to send the auth handshake and then less than 10% to send the p2p handshake